tests/v2/integration/resource_quota_test.go

```diff
@@ -0,0 +1,284 @@
+package integration
+
+import (
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3quotas "github.com/rancher/rancher/pkg/controllers/managementuser/resourcequota"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/resourcequotas"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ResourceQuotaSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+}
+
+func (s *ResourceQuotaSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *ResourceQuotaSuite) SetupSuite() {
+	testSession := session.NewSession(s.T())
+	s.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+	s.client = client
+}
+
+func (s *ResourceQuotaSuite) TestCreateNamespaceWithQuotaInProject() {
+	subSession := s.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := s.client.WithSession(subSession)
+	require.NoError(s.T(), err)
+
+	projectLimit := &management.ResourceQuotaLimit{
+		LimitsCPU: "500m",
+	}
+	namespaceDefaultLimit := &management.ResourceQuotaLimit{
+		LimitsCPU: "200m",
+	}
+
+	projectConfig := &management.Project{
+		ClusterID: "local",
+		Name:      "TestProject",
+		ResourceQuota: &management.ProjectResourceQuota{
+			Limit: projectLimit,
+		},
+		NamespaceDefaultResourceQuota: &management.NamespaceResourceQuota{
+			Limit: namespaceDefaultLimit,
+		},
+	}
+	testProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(s.T(), err)
+
+	namespace, err := namespaces.CreateNamespace(client, "ns1", "", map[string]string{}, map[string]string{}, testProject)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), namespace)
+
+	quotas, err := resourcequotas.ListResourceQuotas(client, "local", "ns1", metav1.ListOptions{})
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), quotas)
+	require.Lenf(s.T(), quotas.Items, 1, "Expected 1 quota in a new namespace, but got %d", len(quotas.Items))
+
+	limit := &v3.ResourceQuotaLimit{
+		LimitsCPU: "200m",
+	}
+
+	resourceList := quotas.Items[0].Spec.Hard
+	rl, err := v3quotas.ConvertProjectResourceLimitToResourceList(limit)
```

Comment:
Instead of changing this function to exported just for use in this test, is it possible to create a native resource list object to compare to rather than converting a v3 object?

Reply:
Agreed, fixed!